CRM/Contribute/Form/ContributionPage/Settings.php

```diff
@@ -348,8 +348,16 @@ public function postProcess() {
 
     foreach ($ufJoinParams as $index => $ufJoinParam) {
       if (!empty($params[$index])) {
-        // first delete all past entries
-        CRM_Core_BAO_UFJoin::deleteAll($ufJoinParam);
+        // Look for an existing entry
+        $ufJoinDAO = new CRM_Core_DAO_UFJoin();
+        $ufJoinDAO->module = $ufJoinParam['module'];
+        $ufJoinDAO->entity_id = $ufJoinParam['entity_id'];
+        $ufJoinDAO->find(TRUE);
```

Comment:
Is it possible that soft-credit profiles could be used for something other than contribution pages? If so, you'd want to take entity_table into account. It used to before with the deleteAll. Otherwise this might match on multiple and might update the wrong one.

Reply:
I think the combo would be surprising, but I added an extra check, just in case.
However, I only did a quick code edit, need to re-test.